src/rewards/voters.rs

```diff
@@ -33,14 +37,65 @@ pub fn calculate_stake<'address>(
 pub fn calculate_reward_share<'address>(
     total_stake: u64,
     stake_per_voter: &HashMap<&'address Address, u64>,
+    threshold_addresses: &AddressesVoteCount,
+    threshold: u64,
 ) -> HashMap<&'address Address, U64F64> {
     stake_per_voter
         .iter()
-        .map(|(k, v)| (*k, fixed::types::U64F64::from_num(*v) / total_stake as u128))
+        .map(|(k, v)| {
+            let reward = if *threshold_addresses
+                .get(k)
+                .unwrap_or_else(|| panic!("Missing address {} in vote count map", k))
+                >= threshold
+            {
+                U64F64::from_num(*v) / total_stake as u128
+            } else {
+                U64F64::default()
+            };
+            (*k, reward)
+        })
         .collect()
 }
 
 /// get the proportional reward from a share and total rewards amount
 pub fn reward_from_share(share: U64F64, total_reward: u64) -> fixed::types::U64F64 {
     fixed::types::U64F64::from_num(total_reward) * share
 }
+
+pub type VoteCount = HashMap<String, u64>;
+pub type AddressesVoteCount = HashMap<Address, u64>;
```

Comment:
What about implementing `Deserialize` on `AddressesVoteCount` not to have a type (`VoteCount`) which is actually never used in practice?

Reply:
Good idea. Will do.

Reply:
Actually, I think it would be helpful to keep like this. Reason is that when debugging could be clearer that we rematch the hexlified identifiers to the Addresses themselves. It may be handier to do it in the deserialize implementation but also maybe more confusing.

Reply:
> Reason is that when debugging could be clearer that we rematch the hexlified identifiers to the Addresses themselves

Could be solvable with the custom `Debug` impl.

Reply:
No really, I mean in the code flow. It is an explicit call instead of some magic behind the deserialization itself.